### DISCUSSION TITLE
useMutation: `enabled` prop

### DISCUSSION THREAD

**Discussion by bazuka5801** [TanStack/query#5922] [Q&A] [answered]
I'm making a mobile app with tan stack query, yeah, I'm exist.
And with offline x2)
P.s. If you doing something that, I will appreciate for your emoji.

Steps:
0. Turn off wife
1. mutate(...) something
2. Close the app (with queryCacheProvider)
3. Turn on wife
4. Open the app

First `Query` executes `pausedMutations`, but the token (for paused mutations) will be obtained after all `pausedMutations` in an additional mutation like `auth.getToken`.

I want to stop some mutations until the application gets the token, and only then execute them.

**Comment by bazuka5801** [comment, marked as answer]
I think this may be a solution to the problem:
```ts
watch(
  queryClient.isRestoring,
  async (isRestoring) => {
    if (!isRestoring) {
      // Wait auth....
      console.log('Waiting auth for resume paused mutations...')
      await until(isAuthenticated).toBe(true)
      queryClient.resumePausedMutations()
      console.log('Resume', queryClient)
    }
  },
  { immediate: true }
)
```

**Comment by Mizynets** [comment]
@bazuka5801 Hi!, 
Use  `setMutationDefaults`

```
// we need a default mutation function so that paused mutations can resume after a page reload
queryClient.setMutationDefaults(movieKeys.all(), {
  mutationFn: async ({ id, comment }) => {
    // to avoid clashes with our optimistic update when an offline mutation continues
    await queryClient.cancelQueries({ queryKey: movieKeys.detail(id) })
    return api.updateMovie(id, comment)
  },
})
```

codes from [example](https://github.com/richardszegh/tanstack-query-offline-example/blob/main/src/App.jsx)
